hera_pspec/container.py

```diff
@@ -0,0 +1,305 @@
+import numpy as np
+import h5py
+from hera_pspec.uvpspec import UVPSpec
+import hera_pspec.version as version
+
+class PSpecContainer(object):
+    """
+    Container class for managing multiple UVPSpec objects.
+    """
+    
+    def __init__(self, filename, mode='r'):
+        """
+        Manage a collection of UVPSpec objects that are stored in a structured 
+        HDF5 file. 
+        
+        Parameters
+        ----------
+        filename : str
+            Path to HDF5 file to store power spectra in.
+        
+        mode : str
+            Whether to load the HDF5 file as read/write ('rw') or read-only 
+            ('r'). If 'rw' is specified and the file doesn't exist, an empty 
+            one will be created.
+        """
+        self.filename = filename
+        self.mode = mode
+        if mode not in ['r', 'rw']:
+            raise ValueError("Must set mode to either 'r' or 'rw'.")
+        
+        # Open file ready for reading and/or writing
+        self.data = None
+        self._open()
+    
+    
+    def _open(self):
+        """
+        Open HDF5 file ready for reading/writing.
+        """
+        # Convert user-specified mode to a mode that HDF5 recognizes. We only 
+        # allow non-destructive operations!
+        mode = 'a' if self.mode == 'rw' else 'r'
+        self.data = h5py.File(self.filename, mode)
+        if self.mode == 'rw':
+            self._update_header()
+    
+    
+    def _store_pspec(self, pspec_group, ps):
+        """
+        Store a UVPSpec object as group of datasets within the HDF5 file.
+        
+        Parameters
+        ----------
+        pspec : HDF5 group
+            HDF5 group to store power spectrum data in.
+        
+        ps : UVPSpec
```

Comment:
a little pedantic, but to be consistent with other parts of the repo, can we use the variable name `uvp` (instead of `ps`) to refer to an instance of a UVPSpec object?